core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -174,22 +177,53 @@ class GroupCoordinator(val brokerId: Int,
       } else {
         val newMemberId = clientId + "-" + group.generateMemberIdSuffix
 
-        if (requireKnownMemberId) {
-          // If member id required, register the member in the pending member list
-          // and send back a response to call for another join group request with allocated member id.
-          group.addPendingMember(newMemberId)
-          addPendingMemberExpiration(group, newMemberId, sessionTimeoutMs)
-          responseCallback(joinError(newMemberId, Errors.MEMBER_ID_REQUIRED))
+        if (group.hasStaticMember(groupInstanceId)) {
+          val oldMemberId = group.getStaticMemberId(groupInstanceId)
+
+          if (group.is(Stable)) {
+            debug("Static member $groupInstanceId with unknown member id rejoins, assigning new member id $newMemberId. " +
```

Comment:
nit: I think it worth being an `info` since this should not happen frequently and hence each time it happens we should pay attention.

Reply:
Also: better include the groupInstanceId as well?

Reply:
Sounds good